pennylane/circuit_graph.py

```diff
@@ -296,16 +296,14 @@ def ancestors(self, ops):
             ops (Iterable[Operator]): set of operators in the circuit
 
         Returns:
-            set[Operator]: ancestors of the given operators
+            list[Operator]: ancestors of the given operators
```

Comment:
So here the idea is that use of `set` is an indirect use of `__hash__`?

And our new way of calculating `hash` will be incompatible with what we need from a `set`?

Reply:
Yep. `CircuitGraph` needs to return all unique objects, but it will only return operators that aren't equal if we use a set.